src/terminal/Line.cpp

```diff
@@ -29,8 +29,13 @@ template <typename Cell>
 typename Line<Cell>::InflatedBuffer Line<Cell>::reflow(ColumnCount _newColumnCount)
 {
     using crispy::Comparison;
+    if (isTrivialBuffer())
+    {
+        auto& buffer = trivialBuffer();
+        buffer.displayWidth = _newColumnCount;
```

Comment:
I think this optimization can only be done if the text on that line is smaller or equal to the `_newColumnCount`. Otherwise the slow path could be taken.

What would happen now is that if you shrink only the columns, they'll not be wrapped around but simply not shown whatever would overflow (or wrap around).